src/SFML/Window/macOS/WindowImplCocoa.mm

```diff
@@ -255,87 +239,50 @@ void showMouseCursor()
 ////////////////////////////////////////////////////////////
 void WindowImplCocoa::mouseDownAt(Mouse::Button button, int x, int y)
 {
-    Event event;
-    event.type               = Event::MouseButtonPressed;
-    event.mouseButton.button = button;
-    event.mouseButton.x      = x;
-    event.mouseButton.y      = y;
-    scaleOutXY(event.mouseButton, m_delegate);
-
-    pushEvent(event);
+    scaleOutWidthHeight(x, y, m_delegate);
+    pushEvent(Event::MouseButtonPressed{button, {x, y}});
 }
 
 
 ////////////////////////////////////////////////////////////
 void WindowImplCocoa::mouseUpAt(Mouse::Button button, int x, int y)
 {
-    Event event;
-    event.type               = Event::MouseButtonReleased;
-    event.mouseButton.button = button;
-    event.mouseButton.x      = x;
-    event.mouseButton.y      = y;
-    scaleOutXY(event.mouseButton, m_delegate);
-
-    pushEvent(event);
+    scaleOutWidthHeight(x, y, m_delegate);
+    pushEvent(Event::MouseButtonReleased{button, {x, y}});
 }
 
 
 ////////////////////////////////////////////////////////////
 void WindowImplCocoa::mouseMovedAt(int x, int y)
 {
-    Event event;
-    event.type        = Event::MouseMoved;
-    event.mouseMove.x = x;
-    event.mouseMove.y = y;
-    scaleOutXY(event.mouseMove, m_delegate);
-
-    pushEvent(event);
+    scaleOutWidthHeight(x, y, m_delegate);
+    pushEvent(Event::MouseMoved{{x, y}});
 }
 
 ////////////////////////////////////////////////////////////
 void WindowImplCocoa::mouseWheelScrolledAt(float deltaX, float deltaY, int x, int y)
 {
-    Event event;
-
-    event.type                   = Event::MouseWheelScrolled;
-    event.mouseWheelScroll.wheel = Mouse::VerticalWheel;
-    event.mouseWheelScroll.delta = deltaY;
-    event.mouseWheelScroll.x     = x;
-    event.mouseWheelScroll.y     = y;
-    scaleOutXY(event.mouseWheelScroll, m_delegate);
-    pushEvent(event);
+    scaleOutWidthHeight(x, y, m_delegate);
+    pushEvent(Event::MouseWheelScrolled{Mouse::VerticalWheel, deltaY, {x, y}});
 
-    event.type                   = Event::MouseWheelScrolled;
-    event.mouseWheelScroll.wheel = Mouse::HorizontalWheel;
-    event.mouseWheelScroll.delta = deltaX;
-    event.mouseWheelScroll.x     = x;
-    event.mouseWheelScroll.y     = y;
-    scaleOutXY(event.mouseWheelScroll, m_delegate);
-    pushEvent(event);
+    scaleOutWidthHeight(x, y, m_delegate);
```

Comment:
Bug here: this is scaling the same variables a second time. You can just remove the line.